tools/rewriter/SourceRewriter.cpp

```diff
@@ -614,6 +614,11 @@ class FnPtrExpr : public RefactoringCallback {
     auto *fn_ptr_expr = result.Nodes.getNodeAs<clang::DeclRefExpr>("fnPtrExpr");
     assert(fn_ptr_expr != nullptr);
 
+    auto annotation = fn_ptr_expr->getDecl()->getAttr<clang::AnnotateAttr>();
+    if (annotation && annotation->getAnnotation() == SKIP_WRAP_ATTR) {
+        return;
+    }
```

Comment:
Is this needed for this change? Just wanted to make sure it was related and not accidental.

Reply:
yeah, but it's not obvious why so I should've update the documentation in ia2.h. The attribute is added to the function declaration in the `Test` macro and this branch makes it so that expressions that refer to that dec (e.g. the RHS in `.test = fake_criterion_##suite##_##name,`) are not rewritten. In the case of `Test` it wasn't being rewritten since it's expanded from a macro, but the rewriter was printing out warnings about needing to manually rewrite that spot.

Reply:
Ideally I would've like to have `IA2_{BEGIN,END}_NO_WRAP` apply to only things between those macros but I don't think we can annotate expressions so I don't see an obvious way to do this.

Reply:
Actually this pass already called `ignore_function` so I'll remove it since it wasn't necessary. I updated the comment on the `Test` macro definition to explain why the NO_WRAP is necessary there but I'll leave updating the `NO_WRAP` documentation to a separate change after checking exactly what it applies to (it doesn't apply to function pointers between the macros, but it does apply to pointers that reference functions defined between the macros).